plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -3691,4 +3749,24 @@ private static Optional<String> getQueryId(Table table)
     {
         return Optional.ofNullable(table.getParameters().get(PRESTO_QUERY_ID_NAME));
     }
+
+    private static class GetInsertCommitVersionExecutionContext
```

Comment:
Will we introduce another context class for other statements, UPDATE/MERGE/DELETE?

Reply:
Changed the class for now to `AtomicReference` because we just need a holder for the read version of the table.